homeassistant/components/diagnostics/__init__.py

```diff
@@ -118,6 +130,69 @@ async def get(  # pylint: disable=no-self-use
             body=json_data,
             content_type="application/json",
             headers={
-                "Content-Disposition": f'attachment; filename="{config_entry.domain}-{config_entry.entry_id}.json"'
+                "Content-Disposition": f'attachment; filename="{d_type}-{filename}.json"'
+            },
+        )
+
+
+class DownloadDiagnosticsSubConfigEntryView(http.HomeAssistantView):
+    """Download diagnostics from an object underneath config entry view."""
+
+    url = "/api/diagnostics/config_entry/{config_entry_id}/{sub_type}/{sub_id}"
```

Comment:
it needs to be optional. We should still be able to et config entry data too. Use extra_urls to add extra mappers.